In [3]:
import pandas as pd

In [4]:
data=pd.read_csv('Sales_add.csv')
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [5]:
data.shape

(22, 5)

In [6]:
data.nunique()

Month                             22
Region                             3
Manager                            3
Sales_before_digital_add(in $)    22
Sales_After_digital_add(in $)     22
dtype: int64

# 1. Whether there is any increase in sales after stepping into digital marketing.

Here we can conduct a paired sample t-test since we want to compare the mean of sales before and after digital marketing

H0 : There is increase in sales after stepping into digital marketing

H1 : There is no increase in sales after stepping into digital marketing

In [10]:
import scipy.stats as stats

In [12]:
stats.ttest_rel(data['Sales_before_digital_add(in $)'],data['Sales_After_digital_add(in $)'])

Ttest_relResult(statistic=-12.09070525287017, pvalue=6.336667004575778e-11)

Here the p value 6.336667004575778e-11 is less than assumed alpha value 0.05
Therefore we cannot reject the null hypothesis

INFERENCE: There is increase in sales after stepping into digital marketing

# 2. Check whether there is any dependency between the features “Region” and “Manager”

We can use chi-square test to check for the dependancy between 'Region' and 'Manager'

H0: There is no relationship between Region and Manager

H1: There is relationship between Region and Manager

In [17]:
obs=pd.crosstab(data['Region'],data['Manager'], margins=True)

In [18]:
obs

Manager     Manager - A  Manager - B  Manager - C  All
Region                                                
Region - A            4            3            3   10
Region - B            4            1            2    7
Region - C            1            3            1    5
All                   9            7            6   22

In [29]:
from scipy.stats import chi2_contingency
stats.chi2_contingency(obs)

(0.6921428571428572,
 0.8750504542399697,
 3,
 array([[ 4.70588235,  2.35294118,  2.94117647, 10.        ],
        [ 3.29411765,  1.64705882,  2.05882353,  7.        ]]))

Here our p value is 0.8750504542399697 which is greater than assumed alpha value 0.05 therefore we accept the null hypothesis

INFERENCE: There is no relationship between the features 'Region' and 'Manager'